### DISCUSSION TITLE
Read You Release Protocols

### DISCUSSION THREAD

**Discussion by mbestavros** [ReadYouApp/ReadYou#501] [General]
As of today (2024-01-09), there are now three people capable of managing the repo:

- @Ashinch, the project's creator and leader
- @JunkFood02 and myself, who were just added as collaborators.

We should discuss how a release gets done given that we have multiple people with capability to do it.

## My proposal

@Ashinch should generally have final say on release decisions, since ReadYou is their project, but when unavailable or busy, we still definitely want releases to happen. To facilitate that, let's follow these rules:

- @Ashinch can cut a release on his own, no permission needed
- If @Ashinch is unable to cut a release, one of the other contributors can initiate a release, but only after receiving approval from one of the other collaborators (via an issue)

For example, if I wanted to kick off a `0.9.9` release, I'd create an issue with some brief rationale for why I think it's needed. I'd tag both @Ashinch and @JunkFood02, and once one of them gives their explicit approval, I'll go ahead and cut the release.

I've actually done just that here, as an example/putting this plan in action: https://github.com/Ashinch/ReadYou/issues/500

@Ashinch @JunkFood02 Let me know what you think of this plan!

**Comment by mbestavros** [comment]
@Ashinch Can you tell me which APK file you use for releases? I understand how to create a new tag and mark a new release, but from what I can see none of the GH Actions automatically upload artifacts to the release page.

Do you just take a build artifact from the `build_commit.yaml` workflow?

**Reply by Ashinch** [reply]
@mbestavros 

My typical release process involves the following steps:

1. Merge translation pull requests from Weblate.
2. Make a new commit, including changes to the version number and F-Droid's changelog.
3. Create a tag based on the second step.
4. Wait for the GitHub workflow "build_commit" to complete the build for the new version.
5. Retrieve the APK file from the artifacts.
6. Draft a new release page and upload the new APK file.

**Reply by mbestavros** [reply]
@Ashinch Thank you! I'll put together `0.9.9` soon!

**Reply by mbestavros** [reply]
The release is out! https://github.com/Ashinch/ReadYou/releases/tag/0.9.9

**Reply by mbestavros** [reply]
@Ashinch @JunkFood02 I wrote up a wiki page with the release process, for posterity: https://github.com/Ashinch/ReadYou/wiki/Read-You-Release-Process
